File: Java/happyNumber.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> abc= new HashSet<>();
        while(n!=1 && !abc.contains(n)){
            int k= n;
            int sum= 0;

            while(k>0){
                sum+= Math.pow(k%10,2);
                k/=10;
            }
            System.out.println(sum);
            abc.add(n);
            n=sum;
        }

        return n==1;
    }
}

File: Python/happyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        abc= set()

        while n!=1 and n not in abc:
            k= n
            digitSum= 0
            while k>0:
                digitSum+= math.pow(k%10,2)
                k//=10

            abc.add(n)
            n= digitSum
        
        return n==1